sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+"""
+This module can be used to solve 2D beam bending problems with
+singularity functions in mechanics.
+
+"""
+
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, Eq, diff
+from sympy.solvers import linsolve
+from sympy.printing import sstr
+from sympy.functions import SingularityFunction
+from sympy.core import sympify
+from sympy.integrals import integrate
+
+
+class Beam(object):
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending.
+    Beams are characterized by their profile (Second moment of area),
+    their length, and their material.
+
+    Parameters
+    ==========
+    length : Sympifyable
+        A SymPy expression representing the Beam's length.
+    elastic_modulus : Sympifyable
+        A SymPy expression representing the Beam's Modulus of Elasticity.
+        It is a measure of the stiffness of the Beam material.
+    second_moment : Sympifyable
+        A SymPy expression representing the Beam's Second moment of area.
+        It is a geometrical property of an area which reflects how its
+        points are distributed with respect to its neutral axis.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import Beam, PointLoad, DistributedLoad
+    >>> from sympy import Symbol
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> Load_1 = PointLoad(location = 0, value = -3)
+    >>> Load_2 = PointLoad(location = 4, value = -9)
+    >>> Load_3 = DistributedLoad(value = 6, start = 2, order = 0)
+    >>> b.apply_loads(Load_1, Load_2, Load_3)
+    >>> b.apply_boundary_conditions(deflection = [(4, 0)])
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load_distribution()
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
```

Comment:
Pretty printing is not working in the docstrings. 
In terminal : 
```py
In [2]: from sympy.physics.continuum_mechanics.beam import Beam, PointLoad, DistributedLoad

In [3]: from sympy import Symbol, Piecewise, pprint

In [4]: E = Symbol('E')

In [5]: I = Symbol('I')

In [6]: b = Beam(4, E, I)

In [7]: Load_1 = PointLoad(location = 0, value = -3)

In [8]: Load_2 = PointLoad(location = 4, value = -9)

In [9]: Load_3 = DistributedLoad(value = 6, start = 2, order = 0)

In [10]: b.apply_loads(Load_1)

In [11]: b.apply_loads(Load_1)

In [12]: b.apply_loads(Load_2, Load_3)

In [13]: b.load_distribution()
Out[13]: -6*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)

In [14]: pprint(b.load_distribution())
       -1            0            -1
- 6⋅<x>   + 6⋅<x - 2>  - 9⋅<x - 4>  

```
In docstrings : 
```py
File "sympy/physics/continuum_mechanics/beam.py", line 51, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    pprint(b.load_distribution())
Expected:
    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
Got:
    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*Singular
    ityFunction(x, 4, -1)

```